src/script/script.h

```diff
@@ -68,142 +68,153 @@ std::vector<unsigned char> ToByteVector(const T& in)
 
     // splice ops
-    OP_CAT = 0x7e,
-    OP_SUBSTR = 0x7f,
-    OP_LEFT = 0x80,
-    OP_RIGHT = 0x81,
-    OP_SIZE = 0x82,
+    OP_CAT = 0x7e,    // mark transaction invalid even when occurring in an unexecuted branch
+    OP_SUBSTR = 0x7f, // mark transaction invalid even when occurring in an unexecuted branch
+    OP_LEFT = 0x80,   // mark transaction invalid even when occurring in an unexecuted branch
+    OP_RIGHT = 0x81,  // mark transaction invalid even when occurring in an unexecuted branch
+    OP_SIZE = 0x82,   // push the length of top stack item (not pop the top element whose size is inspected)
 
     // bit logic
-    OP_INVERT = 0x83,
-    OP_AND = 0x84,
-    OP_OR = 0x85,
-    OP_XOR = 0x86,
-    OP_EQUAL = 0x87,
-    OP_EQUALVERIFY = 0x88,
-    OP_RESERVED1 = 0x89,
-    OP_RESERVED2 = 0x8a,
+    OP_INVERT = 0x83,      // mark transaction invalid even when occurring in an unexecuted branch
+    OP_AND = 0x84,         // mark transaction invalid even when occurring in an unexecuted branch
+    OP_OR = 0x85,          // mark transaction invalid even when occurring in an unexecuted branch
+    OP_XOR = 0x86,         // mark transaction invalid even when occurring in an unexecuted branch
+    OP_EQUAL = 0x87,       // pop two top stack items and push "\x01" if they are equal, otherwise ""
+    OP_EQUALVERIFY = 0x88, // execute OP_EQUAL, then OP_VERIFY afterward
+    OP_RESERVED1 = 0x89,   // mark transaction invalid
+    OP_RESERVED2 = 0x8a,   // mark transaction invalid
 
     // numeric
-    OP_1ADD = 0x8b,
-    OP_1SUB = 0x8c,
-    OP_2MUL = 0x8d,
-    OP_2DIV = 0x8e,
-    OP_NEGATE = 0x8f,
-    OP_ABS = 0x90,
-    OP_NOT = 0x91,
-    OP_0NOTEQUAL = 0x92,
-
-    OP_ADD = 0x93,
-    OP_SUB = 0x94,
-    OP_MUL = 0x95,
-    OP_DIV = 0x96,
-    OP_MOD = 0x97,
-    OP_LSHIFT = 0x98,
-    OP_RSHIFT = 0x99,
-
-    OP_BOOLAND = 0x9a,
-    OP_BOOLOR = 0x9b,
-    OP_NUMEQUAL = 0x9c,
-    OP_NUMEQUALVERIFY = 0x9d,
-    OP_NUMNOTEQUAL = 0x9e,
-    OP_LESSTHAN = 0x9f,
-    OP_GREATERTHAN = 0xa0,
-    OP_LESSTHANOREQUAL = 0xa1,
-    OP_GREATERTHANOREQUAL = 0xa2,
-    OP_MIN = 0xa3,
-    OP_MAX = 0xa4,
-
-    OP_WITHIN = 0xa5,
+    OP_1ADD = 0x8b,      // add 1 to the top stack item
+    OP_1SUB = 0x8c,      // subtract 1 from the top stack item
+    OP_2MUL = 0x8d,      // mark transaction invalid even when occurring in an unexecuted branch
+    OP_2DIV = 0x8e,      // mark transaction invalid even when occurring in an unexecuted branch
+    OP_NEGATE = 0x8f,    // multiply the top stack item by -1
+    OP_ABS = 0x90,       // replace top stack item by its absolute value
+    OP_NOT = 0x91,       // replace top stack item by "\x01" if its value is 0, otherwise ""
+    OP_0NOTEQUAL = 0x92, // replace top stack item by "" if its value is 0, otherwise by "\x01"
+
+    OP_ADD = 0x93,    // pop two top stack items and push their sum
+    OP_SUB = 0x94,    // pop two top stack items and push the second minus the top
```

Comment:
`pop two` doesn't quite work (IMO).

it makes the order somewhat ambiguous.

for example:

`pop pop add push` is more correct. no? 

Being more explicit in this way will make the `OP_SUB` make more sense, no?.

`pop pop subtract push`


for `OP_ADD` it isn't much of an issue.

`` 1 + 1 = 2 ``

`` 1 + -1 = 0 ``

`` -1 + 1 = 0 ``




for `OP_SUB` it may be an issue.

``1 - 1 = 0``

``1 - -1 = 0``

``-1 - 1 = -2``

*in common arithmetic notation